testdata/generated/categories-experimental.json

```diff
@@ -31,22 +31,25 @@
   },
   {
     "id": "datastore",
-    "title": "Datastore",
+    "title": "Database",
     "count": 3
   },
   {
     "id": "message_queue",
-    "title": "Message Queue",
-    "count": 1
+    "title": "Message Broker",
+    "count": 1,
+    "subcategory_of": "infrastructure"
   },
   {
     "id": "monitoring",
     "title": "Monitoring",
-    "count": 3
+    "count": 3,
+    "subcategory_of": "observability"
   },
   {
     "id": "web",
-    "title": "Web",
-    "count": 3
+    "title": "Web Server",
+    "count": 3,
+    "subcategory_of": "infrastructure"
```

Comment:
@kpollich with this API response, Fleet is not going to be able to get the title of the parent category, should I change this to something like this?
```suggestion
    "parent_id": "infrastructure",
    "parent_name": "Infrastructure"
```

An alternative could be to include the parent category in the response, but this could affect current versions of kibana that would receive categories with count 0.

This also makes me wonder if we want to modify the search APIs, to include packages of subcategories, when searching for a main category. That would for example be to include packages of the `web` subcategory, when looking for the `integration` category.

Reply:
> with this API response, Fleet is not going to be able to get the title of the parent category, should I change this to something like this?

It seems like the issue is that integrations can be placed in a subcategory even if there are no integrations placed in the parent category, resulting in a case where the parent category is not included in the response. 

So we have a state like this

![image](https://user-images.githubusercontent.com/6766512/204870731-f79084dc-e2fa-4681-92f5-7712211c37c5.png)

I think returning the `parent_name` would be most helpful here, as it allows the UI to render the `Infrastructure` category in this case even if it does not directly contain any integrations.

Reply:
PR updated to include the `parent_name` in the API response, @kpollich please take another look.